### PR TITLE
Add document and sample codes for auto_accelerate API

### PR DESCRIPTION
**Pull request by skydoorkai**
### What changes were proposed in this pull request?

Please describe the changes you have made or proposed in this pull request.

### Why are the changes needed?

Explain the purpose or motivation behind these changes. What problem are you trying to solve?

### Does this PR introduce any user-facing change?

Specify whether this pull request introduces any changes that users will directly interact with or notice.

### How was this patch tested?

Detail the testing process you have undertaken to ensure the changes in this pull request are valid and working as intended.

### CODE REVIEW

File: atorch/examples/auto_accelerate/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+## Examples
+
+Training toy model in fully-automatic mode.
+```
+python train.py --model_type toy
+```
+with distributed training.
+
+```
+python -m atorch.distributed.launch --nproc_per_node 2 train.py --model_type toy --distributed
```
Comment (workingloong) [inline atorch/examples/auto_accelerate/README.md:57]: Why do we not use `python -m atorch.distributed.run`?

File: atorch/examples/auto_accelerate/data.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    ):
+        self.vocab_size = vocab_size
+        self.max_source_positions = max_source_positions
+        self.total_samples = total_samples
+        self.sizes = [self.max_source_positions] * self.total_samples
+
+    def __getitem__(self, index):
+        length = self.sizes[index]
+        source = torch.randint(1, self.vocab_size, (length,))
+        target = source.clone()
```
Comment (workingloong) [inline atorch/examples/auto_accelerate/data.py:56]: Should we need to left shift the source with one token for the target?

File: atorch/examples/auto_accelerate/README.md (hunk 1 above)
Comment (skydoorkai) [reply atorch/examples/auto_accelerate/README.md]: Changed to atorch.distributed.run.

**Review by workingloong — APPROVED** [review]
LGTM

**Pull request merged**